app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/EmptyState.kt

```diff
@@ -0,0 +1,201 @@
+import com.hedvig.android.design.system.hedvig.EmptyStateDefaults.EmptyStateIconStyle.INFO
+import com.hedvig.android.design.system.hedvig.EmptyStateDefaults.EmptyStateIconStyle.NO_ICON
+import com.hedvig.android.design.system.hedvig.EmptyStateDefaults.EmptyStateIconStyle.SUCCESS
+import com.hedvig.android.design.system.hedvig.icon.BankId
+import com.hedvig.android.design.system.hedvig.icon.CheckFilled
+import com.hedvig.android.design.system.hedvig.icon.HedvigIcons
+import com.hedvig.android.design.system.hedvig.icon.InfoFilled
+import com.hedvig.android.design.system.hedvig.icon.WarningFilled
+import com.hedvig.android.design.system.hedvig.tokens.EmptyStateTokens
+
+@Composable
+fun EmptyState(
+  text: String,
+  description: String?,
+  modifier: Modifier = Modifier,
+  iconStyle: EmptyStateIconStyle = EmptyStateDefaults.iconStyle,
+  buttonStyle: EmptyStateButtonStyle = EmptyStateDefaults.buttonStyle,
+) {
+  Column(
+    modifier = modifier
+      .background(emptyStateColors.containerColor)
+      .padding(horizontal = 16.dp),
+    horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    Spacer(Modifier.height(24.dp))
+    EmptyStateIcon(iconStyle)
+    HedvigText(
+      text = text,
+      style = EmptyStateDefaults.textStyle.copy(
+        lineBreak = LineBreak.Heading,
+      ),
+      color = emptyStateColors.textColor,
+      textAlign = TextAlign.Center,
+    )
+    if (description != null) {
+      HedvigText(
+        description,
+        style = EmptyStateDefaults.textStyle.copy(
+          lineBreak = LineBreak.Heading,
+        ),
+        color = emptyStateColors.descriptionColor,
+        textAlign = TextAlign.Center,
+      )
+    }
+    when (buttonStyle) {
+      is Button -> {
+        Spacer(Modifier.height(24.dp))
+        HedvigButton(
+          onClick = buttonStyle.onButtonClick,
+          enabled = true,
+          buttonStyle = Primary,
+          buttonSize = Medium,
+        ) {
+          HedvigText(buttonStyle.buttonText)
+        }
+      }
+      NoButton -> {}
+    }
+    Spacer(Modifier.height(24.dp))
+  }
+}
+
+@Composable
+private fun ColumnScope.EmptyStateIcon(iconStyle: EmptyStateIconStyle) {
+  val sizeModifier = Modifier.size(EmptyStateTokens.IconSize)
+  when (iconStyle) {
+    ERROR -> {
+      Icon(
+        HedvigIcons.WarningFilled,
+        null,
+        tint = emptyStateColors.errorIconColor,
+        modifier = sizeModifier,
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+
+    INFO -> {
+      Icon(
+        HedvigIcons.InfoFilled,
+        null,
+        tint = emptyStateColors.infoIconColor,
+        modifier = sizeModifier,
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+
+    SUCCESS -> {
+      Icon(
+        HedvigIcons.CheckFilled,
+        null,
+        tint = emptyStateColors.successIconColor,
+        modifier = sizeModifier,
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+
+    BANK_ID -> {
+      Icon(
+        HedvigIcons.BankId,
+        null,
```

Comment:
Does this use the LocalContentColor as tint? Does it end up showing correctly for light and dark mode properly?

Reply:
Yes, it's LocalContentColor.current, the color is inverted in the dark mode (as it should be)